From patchwork Thu Jan  6 22:29:01 2011
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [3/3] xfree86: Fix the sdk headers to be multilib-safe
Date: Fri, 07 Jan 2011 06:29:01 -0000
From: Adam Jackson <ajax@redhat.com>
X-Patchwork-Id: 3576
Message-Id: <1294352941-16508-4-git-send-email-ajax@redhat.com>
To: xorg-devel@lists.x.org

Use _LP64 (verified on gcc and sun compilers) instead of _XSERVER64 in
internal header usage, and always define _XSERVER64 for modules if _LP64
is defined.  Prevents differing xorg-server.h between 32 and 64 bit
packages.

Signed-off-by: Adam Jackson <ajax@redhat.com>
Reviewed-by: Alan Coopersmith <alan.coopersmith@oracle.com>

---
include/colormapst.h     |    4 ++--
 include/xorg-server.h.in |    7 ++++---
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/include/colormapst.h b/include/colormapst.h
index 0ba5bd0..b597e2c 100644
--- a/include/colormapst.h
+++ b/include/colormapst.h
@@ -103,12 +103,12 @@ typedef struct _ColormapRec
 {
     VisualPtr	pVisual;
     short	class;		/* PseudoColor or DirectColor */
-#if defined(_XSERVER64)
+#if defined(_LP64)
     short	pad0;
     XID		pad1;
 #endif
     XID		mid;		/* client's name for colormap */
-#if defined(_XSERVER64) && (X_BYTE_ORDER == X_LITTLE_ENDIAN)
+#if defined(_LP64) && (X_BYTE_ORDER == X_LITTLE_ENDIAN)
     XID		pad2;
 #endif
     ScreenPtr	pScreen;	/* screen map is associated with */
diff --git a/include/xorg-server.h.in b/include/xorg-server.h.in
index 76cab16..b10b994 100644
--- a/include/xorg-server.h.in
+++ b/include/xorg-server.h.in
@@ -157,9 +157,6 @@
 /* Name of X server */
 #undef __XSERVERNAME__
 
-/* Define to 1 if unsigned long is 64 bits. */
-#undef _XSERVER64
-
 /* Building vgahw module */
 #undef WITH_VGAHW
 
@@ -187,4 +184,8 @@
 /* X Access Control Extension */
 #undef XACE
 
+#ifdef _LP64
+#define _XSERVER64 1
+#endif
+
 #endif /* _XORG_SERVER_H_ */
